File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        umap = [0]*101
        for i in nums:
            umap[i] += 1;
        
        for i in range(1, len(umap)):
            umap[i] += umap[i-1];

        for i in range(0, len(nums)):
            if nums[i] == 0:
                nums[i] = 0;
            else:
                nums[i] = umap[nums[i] - 1];
        return nums;
        

File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.go
func smallerNumbersThanCurrent(nums []int) []int {
    var count []int = make([]int, 101);
    for i := 0; i < len(nums); i++ {
        count[nums[i]]++;
    }
    for i:= 1; i < len(count); i++ {
        count[i] += count[i-1];
    }
    
    for i, j := range nums{
        if j == 0{
            nums[i] = 0;
        }else{
            nums[i] = count[j-1];
        }   
    }
    return nums;
}